sparkth_mcp/types.py

```diff
@@ -6,17 +6,29 @@ class CourseGenerationPromptRequest(BaseModel):
     course_description: str
 
 
-class AuthenticationError(Exception):
+class BaseError(Exception):
     def __init__(self, status_code, message):
         self.message = message
         self.status_code = status_code
-        super().__init__(f"{message} (status_code={status_code})")
+        super().__init__(self._format_message())
 
+    def _format_message(self):
+        return f"{self.message} (status_code={self.status_code})"
 
-class LMSError(Exception):
-    def __init__(self, method: str, url, status_code, message):
-        self.status_code = status_code
-        self.message = message
-        self.url = url
+
+class AuthenticationError(BaseError):
+    pass
+
+
+class JsonParseError(BaseError):
+    pass
+
+
+class LMSError(BaseError):
+    def __init__(self, method, url, status_code, message):
         self.method = method
-        super().__init__(f"{method} {url}: {message} (status_code={status_code})")
+        self.url = url
+        super().__init__(message, status_code)
```

Comment:
The order of arguments in `super().__init__(`) is swapped.
    # BaseError expects (status_code, message) but you're passing (message, status_code).